Orange/widgets/data/owsave.py

```diff
@@ -1,12 +1,28 @@
 import os.path
+import pathlib
+
+from AnyQt.QtWidgets import QFormLayout
+from AnyQt.QtCore import Qt
 
 from Orange.data.table import Table
+from Orange.data.io import Compression, FileFormat
 from Orange.widgets import gui, widget
 from Orange.widgets.settings import Setting
-from Orange.data.io import FileFormat
 from Orange.widgets.utils import filedialogs
 from Orange.widgets.widget import Input
 
+FILE_TYPES = [
+    ("Tab-delimited (.tab)", ".tab", False),
+    ("Comma-seperated values (.csv)", ".csv", False),
+    ("Pickle (.pkl)", ".pkl", True),
```

Comment:
Again, these are hard coded constants, which could become incorrect after changes elsewhere.
How did you know where to write False and where True? If you checked the attribute SUPPORT_SPARSE_DATA in reader classes it is better to import and reference that here directly...
Maybe even construct the whole triple directly from the class:

```py
FILE_TYPES = [
    ("{} ({})".format(w.DESCRIPTION, w.EXTENSIONS[0]),
     w.EXTENSIONS[0],
     w.SUPPORT_SPARSE_DATA)
    for w in (TabReader, CSVReader, PickleReader)
]
```

Please test the above code and fix if necessary. It was written here as a suggestion without trying it out.

Also, I think the description for PickledReader could be changed to "Pickled Orange data".